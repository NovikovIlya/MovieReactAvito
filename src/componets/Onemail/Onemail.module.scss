.container{
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    padding-top: 150px;

    &__message{
        width: 100%;
        display: flex;
        justify-content: center;
    }
    &__gr{
        width: 100%;
        display: flex;
        justify-content: center;
    }   
}

.theme,.text{
    margin-bottom: 10px;
}
.send{
    width: 100;
    display: flex;
    justify-content: center;
}

.zagr {
    position: absolute;
    top:300px;
    width: 100%;
    &__hed1 {
      display: flex;
      justify-content: center;
    }
  }

@media screen and (max-width:600px) {
    .container{
        padding-top: 25px;
    }
}