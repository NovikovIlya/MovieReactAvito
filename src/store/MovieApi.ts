import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import {
  ChatAllRequest,
  ChatAllResponse,
  EmailReadRequest,
  GetEmailRequest,
  GetEmailResponse,
  GetUsersResponse,
  MovieApiOneType,
  MovieArray,
  Root2,
  RootYts,
  SendMessageRequest,
  TrailerApi,
  UpdateMessageRequest,
  argType,
  getInfoUser,
  login,
  ratingType,
  renameType,
  repassType,
  tokenType,
} from '../types';



export const MovieApi = createApi({
  reducerPath: 'apiMovies',
  baseQuery: fetchBaseQuery({ baseUrl: 'https://www.omdbapi.com' }),
  tagTypes: ['Fetch'],
  endpoints: (builder) => ({
    fetchMovies: builder.query<MovieArray, string>({
      query: (search) => ({
        url: `?apikey=55ce87c0&s=${search}`,
      }),
    }),
    fetchMoviesOne: builder.query<MovieApiOneType, argType>({
      query: (arg) => ({
        url: `?apikey=55ce87c0&i=${arg.id}`,
      }),
    }),
  }),
});


export const trailerApi = createApi({
  reducerPath: 'trailerApi',
  baseQuery: fetchBaseQuery({
    baseUrl: 'https://api.allorigins.win/raw?url=https://api.kinocheck.de/',
  }),
  tagTypes: ['trailerApi'],
  endpoints: (builder) => ({
    fetcTrailer: builder.query<TrailerApi, string>({
      query: (id) => ({
        url: `movies?imdb_id=${id}`,
      }),
    }),
  }),
});

export const fetchCommentApi = createApi({
  reducerPath: 'fetchComment',
  baseQuery: fetchBaseQuery({ baseUrl: 'https://64b7de9021b9aa6eb079301d.mockapi.io/' }),
  tagTypes: ['fetchComment'],
  endpoints: (builder) => ({
    fetchComment: builder.query<Root2[], string>({
      query: (id) => ({
        url: `comment/?imdbid=${id}`,
      }),
    }),
    fetchRating: builder.query<ratingType[], string>({
      query: (id) => ({
        url: `items/?imdbid=${id}`,
      }),
    }),
    AddRating: builder.mutation<any, any>({
      query: (add) => ({
        method: 'POST',
        url: 'items',
        body: add,
      }),
      invalidatesTags: ['fetchComment'],
    }),
    AddComment: builder.mutation<Root2, Root2>({
      query: (add) => ({
        method: 'POST',
        url: 'comment',
        body: add,
      }),
      invalidatesTags: ['fetchComment'],
    }),
  }),
});



export const torrentApi = createApi({
  reducerPath: 'torrent',
  baseQuery: fetchBaseQuery({ baseUrl: 'https://yts.mx/api/v2/' }),
  tagTypes: ['torrent'],
  endpoints: (builder) => ({
    torrentFetch: builder.query<any, string>({
      query: (imdb) => ({
        url: `list_movies.json?query_term=${imdb}`,
      }),
      providesTags:['torrent']
    }),
    similarFetch: builder.query<any, string>({
      query: (genre) => ({
        url: `list_movies.json?genre=${genre}&limit=10`,
      }),
      providesTags:['torrent']
    }),
    fetchMoviesPopular: builder.query<RootYts, string>({
      query: (search) => ({
        url: `list_movies.json?${search}`,
      }),
    }),
  }),
});


export const info = createApi({
  reducerPath: 'Info',
  baseQuery: fetchBaseQuery({ baseUrl: 'https://backmovie.onrender.com/' }),
  tagTypes: ['getuser','info'],
  endpoints: (builder) => ({
    infoApiSet: builder.mutation<getInfoUser, login>({
      query: (add) => ({
        method: 'POST',
        url: 'auth/info',
        body: add,
      }),
      invalidatesTags: ['getuser'],
    }),
    getUserApiSet: builder.mutation<getInfoUser, login>({
      query: (add) => ({
        method: 'POST',
        url: 'auth/user',
        body: add,
      }),
      invalidatesTags: ['info']
    }),
    RegistrApiSet: builder.mutation<login, login>({
      query: (add) => ({
        method: 'POST',
        url: 'auth/registrationNew',
        body: add,
      }),
      invalidatesTags: ['info'],
    }),
    repassApiSet: builder.mutation<repassType, repassType>({
      query: (add) => ({
        method: 'PUT',
        url: 'auth/repassword',
        body: add,
      }),
      invalidatesTags: ['info'],
    }),
    renameApiSet: builder.mutation<any, renameType>({
      query: (add) => ({
        method: 'PUT',
        url: 'auth/rename',
        body: add,
      }),
      invalidatesTags: ['info'],
    }),
    //1-ответ   2-аргумент(тело запроса)
    LoginApiSet: builder.mutation<tokenType, login>({
      query: (add) => ({
        method: 'POST',
        url: 'auth/login',
        body: add,
      }),
      invalidatesTags: ['info']
    }),
    ChatAll: builder.mutation<ChatAllResponse[], ChatAllRequest>({
      query: (add) => ({
        method: 'POST',
        url: 'chatall',
        body: add,
      }),
      invalidatesTags: ['info']
    }),
    emailRead: builder.mutation<any, EmailReadRequest>({
      query: (add) => ({
        method: 'POST',
        url: 'auth/emailReading',
        body: add,
      }),
      invalidatesTags: ['info']
    }),
    getEmail: builder.mutation<GetEmailResponse, GetEmailRequest>({
      query: (add) => ({
        method: 'POST',
        url: 'auth/getemail',
        body: add,
      }),
      invalidatesTags: ['info']
    }),
    getUsers: builder.query<GetUsersResponse[], string>({
      query: () => ({
        url: 'auth/users',
      }),
      providesTags: ['info']
    }),
    sendMessage: builder.mutation<SendMessageRequest[], SendMessageRequest>({
      query: (add) => ({
        method: 'POST',
        url: 'auth/sendmessage',
        body: add,
      }),
      invalidatesTags: ['info']
    }),
    updateMessage: builder.mutation<SendMessageRequest[], UpdateMessageRequest>({
      query: (add) => ({
        method: 'PUT',
        url: 'auth/updatemessage',
        body: add,
      }),
      invalidatesTags: ['info']
    }),
    getMessage: builder.mutation<SendMessageRequest[], GetEmailRequest>({
      query: (add) => ({
        method: 'POST',
        url: 'auth/getmessage',
        body: add,
      }),
      invalidatesTags: ['info']
    }),
    deleteMessage: builder.mutation<SendMessageRequest[], UpdateMessageRequest>({
      query: (add) => ({
        method: 'DELETE',
        url: 'auth/deletemessage',
        body: add,
      }),
      invalidatesTags: ['info']
    }),
  }),
});

export const infoTag = createApi({
  reducerPath: 'infoTag',
  baseQuery: fetchBaseQuery({ baseUrl: 'https://backmovie.onrender.com/auth/' }),
  tagTypes: ['infoTag'],
  endpoints: (builder) => ({
    infoApiSetTwo: builder.query<any, any>({
      query: (add) => ({
        url: 'info'
      }),
      providesTags: ['infoTag'],
    }),
    getUserApiSetTwo: builder.mutation<any, any>({
      query: (add) => ({
        method: 'POST',
        url: 'user',
        body: add,
      }),
      invalidatesTags: ['infoTag']
    })
    
  }),
})



export const auth = createApi({
  reducerPath: 'auth',
  baseQuery: fetchBaseQuery({
    baseUrl: 'https://backmovie.onrender.com/auth/me',
    prepareHeaders: (headers, { getState }) => {
      headers.set('authorization', `Bearer ${window.localStorage.getItem('token')}`);
      return headers;
    },
  }),
  tagTypes: ['auth'],
  endpoints: (builder) => ({
    authApi: builder.query<login, string>({
      query: () => ({
        url: '',
      }),
    }),
  }),
});

export const { useFetchMoviesQuery,useFetchMoviesOneQuery } = MovieApi;
export const { useFetcTrailerQuery } = trailerApi;
export const { useFetchCommentQuery,useFetchRatingQuery ,useAddRatingMutation,useAddCommentMutation} = fetchCommentApi;
export const { useAuthApiQuery, useLazyAuthApiQuery } = auth;

export const { useTorrentFetchQuery,useSimilarFetchQuery,useFetchMoviesPopularQuery } = torrentApi;

export const {
  useInfoApiSetMutation,
  useGetUserApiSetMutation,
  useRegistrApiSetMutation,
  useRepassApiSetMutation,
  useRenameApiSetMutation,
  useLoginApiSetMutation,
  useChatAllMutation,
  useEmailReadMutation,
  useGetEmailMutation,
  useGetUsersQuery,
  useSendMessageMutation,
  useUpdateMessageMutation,
  useGetMessageMutation,
  useDeleteMessageMutation
} = info;

export const { useInfoApiSetTwoQuery,useGetUserApiSetTwoMutation } = infoTag;

