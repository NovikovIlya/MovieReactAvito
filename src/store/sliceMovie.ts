import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import axios from 'axios';

type MyData = {
  oldUsername: string;
  favorites?: any;
  imdbID?: any;
};

export const deleteFavorites = createAsyncThunk(
  'post/deleteFavorites',
  async (dataBody: MyData) => {
    const { data } = await axios.delete('https://backmovie.onrender.com/auth/deleteFavorites', {
      data: { oldUsername: dataBody.oldUsername, imdbID: dataBody.imdbID },
    });
    return data;
  },
);

export const getFavorites = createAsyncThunk('post/getFavorites', async (dataBody: MyData) => {
  const { data } = await axios.post('https://backmovie.onrender.com/auth/getfavorites', {
    oldUsername: dataBody.oldUsername,
  });
  return data;
});

export const addFavorites = createAsyncThunk('post/addFavorites', async (dataBody: any) => {
  const { data } = await axios.post('https://backmovie.onrender.com/auth/addfavorites', {
    oldUsername: dataBody.oldUsername,
    favoritesNew: dataBody.favorites,
  });
  return data;
});

const initialState = {
  films: [],
  value: '',
  empty: false,
  favorite: [],
  darkMode: false,
  render: false,
  avatar: 'https://innostudio.de/fileuploader/images/default-avatar.png',
  dropdown: 'alipay',
  textComment: 'Enter comment',
  favoritesNew: [],
  isLoad: false,
  closed: false,
  emailAll: 0,
  myName: '',
  num: 1
};

export const sliceMovie = createSlice({
  name: 'sliceMovie',
  initialState,
  reducers: {
    changePag: (state, action) => {
      state.num = action.payload;
      // window.scrollTo({ top: 0, left: 0, behavior: 'smooth' });
    },
    setNumReduce: (state, action) => {
      state.num = action.payload;
      // window.scrollTo({ top: 0, left: 0, behavior: 'smooth' });
    },
    addMovie: (state, action) => {
      state.films = action.payload;
    },
    addValue: (state, action) => {
      state.value = action.payload;
    },
    toogleEmpty: (state, action) => {
      state.empty = action.payload;
    },
    addFavorite: (state, action) => {
      if (state.favorite.find((item) => item.imdbID === action.payload.imdbID)) {
        alert('The movie has already been added to your favorites');
        return;
      }
      state.favorite = [...state.favorite, action.payload];
    },
    deleteFavorite: (state, action) => {

      state.favorite = state.favorite.filter((item) => item.imdbID !== action.payload.imdbID);
    },
    darkMode: (state) => {
      state.darkMode = !state.darkMode;
    },
    toggleRender: (state) => {
      state.render = true;
    },
    switchAvatar: (state, action) => {
      state.avatar = action.payload;
    },
    toggleDropdown: (state, action) => {
      state.dropdown = action.payload;
    },
    addTextComment: (state, action) => {
      state.textComment = action.payload;
    },
    deletefavoritesNew: (state, action) => {
      state.favoritesNew = state.favoritesNew.filter((item) => {
        return item.imdbID !== action.payload;
      });
    },
    setClosed: (state, action) => {
      state.closed = action.payload;
    },
    setEmailAll: (state, action) => {
      state.emailAll = action.payload;
    },
    setMyName: (state,action ) =>{
      state.myName = action.payload
    }
  },
  extraReducers: (builder) => {
    builder.addCase(getFavorites.pending, (state) => {
      state.favoritesNew = [];
      state.isLoad = true;
    });
    builder.addCase(getFavorites.fulfilled, (state, action) => {
      state.favoritesNew = action.payload;
      state.isLoad = false;
    });
    builder.addCase(getFavorites.rejected, (state) => {
      state.favoritesNew = [];
      state.isLoad = false;
    });
  },
});

export const {
  addMovie,
  addValue,
  toogleEmpty,
  addFavorite,
  deleteFavorite,
  darkMode,
  toggleRender,
  switchAvatar,
  toggleDropdown,
  addTextComment,
  deletefavoritesNew,
  setClosed,
  setEmailAll,
  setMyName,
  setNumReduce
} = sliceMovie.actions;
export default sliceMovie.reducer;
