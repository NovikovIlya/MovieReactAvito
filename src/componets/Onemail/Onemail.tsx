import { useState } from 'react';
import styles from './Onemail.module.scss';
import { useGetMessageMutation, useUpdateMessageMutation } from '../../store/MovieApi';
import { useParams } from 'react-router-dom';
import { useEffect } from 'react';
import { useAppSelector } from '../../hooks/redux';
import { Button, Spin } from 'antd';
import { useSendMessageMutation } from '../../store/MovieApi';
import { Modal, Input, message } from 'antd';
import { nanoid } from '@reduxjs/toolkit';
import TextArea from 'antd/es/input/TextArea';
import { SendMessageRequest } from '../../types';

const Onemail = () => {
  const [messageApi, contextHolder] = message.useMessage();
  const [isModalOpen, setIsModalOpen] = useState(false);
  const myName = useAppSelector((state) => state.sliceMovie.myName);
  const [sendMessage] = useSendMessageMutation();
  const [theme, setTheme] = useState('');
  const [text, setText] = useState('');
  const { id, name } = useParams();
  const [getMessage, { data: dataMessage }] = useGetMessageMutation();
  const [updateMessage] = useUpdateMessageMutation();
  const [messageZ, setMessage] = useState<SendMessageRequest | undefined>();

  //hooks
    //Нахождения почты в массиве
  useEffect(() => {
    if (dataMessage) {
      const mess = dataMessage.find((item) => {
        return item.id === id;
      });
      setMessage(mess);
    }
  }, [dataMessage]);

    //Получение почты и Обновление(read)почты
  useEffect(() => {
    getMessage({ username: myName });
    updateMessage({
      username: myName,
      id: id,
    });
  }, [myName]);

  //functions
  const success = () => {
    messageApi.open({
      type: 'success',
      content: 'Message sent',
    });
  };

  const error = () => {
    messageApi.open({
      type: 'error',
      content: 'You cant send an empty message',
    });
  };

  const showModal = () => {
    setIsModalOpen(true);
  };

  const handleOk = () => {
    if (theme.length < 1 || text.length < 1) {
      error();
      return;
    }
    const dataZ = {
      id: nanoid(),
      username: name,
      myname: myName,
      theme: theme,
      text: text,
      date: new Date().toISOString().slice(0, 10).split('-').reverse().join('.'),
      time: new Date().toLocaleTimeString(),
      read: false,
    };
    sendMessage(dataZ);
    success();
    setTheme('');
    setText('');

    setIsModalOpen(false);
  };

  const handleCancel = () => {
    setIsModalOpen(false);
  };

  const handlerTextTheme = ({ target: { value } }) => {
    setTheme(value);
  };

  const handlerTextArea = ({ target: { value } }) => {
    setText(value);
  };

  return (
    <>
      {messageZ ? (
        <div className={styles.container}>
          {contextHolder}
          <div className={styles.container__message}>{messageZ?.text}</div>
          <div className={styles.gr}>
            <Button onClick={showModal}>Reply to message</Button>
          </div>

          <Modal title="Send message" open={isModalOpen} onOk={handleOk} onCancel={handleCancel}>
            <div className={styles.theme}>
              <Input value={theme} name="theme" onChange={handlerTextTheme} placeholder="Theme" />
            </div>
            <div className={styles.text}>
              <TextArea
                value={text}
                name="area"
                onChange={handlerTextArea}
                rows={4}
                placeholder="Message"
                maxLength={60}
              />
            </div>
          </Modal>
        </div>
      ) : (
        <>
          <Spin className={styles.zagr} tip="Loading" size="large">
            <div data-testid="z" className="content" />
          </Spin>
        </>
      )}
    </>
  );
};

export default Onemail;
