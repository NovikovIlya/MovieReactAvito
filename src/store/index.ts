import  sliceMovie  from './sliceMovie';
import { configureStore, combineReducers } from '@reduxjs/toolkit';
import {   MovieApi,auth,fetchCommentApi,info,infoTag,torrentApi,trailerApi } from './MovieApi'; 


import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage'; 
import { setupListeners } from '@reduxjs/toolkit/dist/query';

const rootReducer = combineReducers({
    [MovieApi.reducerPath]: MovieApi.reducer,
    [trailerApi.reducerPath]: trailerApi.reducer,
    [fetchCommentApi.reducerPath]: fetchCommentApi.reducer,
    [auth.reducerPath]: auth.reducer,
    [torrentApi.reducerPath]: torrentApi.reducer,
    [info.reducerPath]: info.reducer,
    [infoTag.reducerPath]: infoTag.reducer,
    sliceMovie,
});


const persistConfig = {
  key: 'root',
  storage,
  blacklist:['toggleDropdown',infoTag.reducerPath,info.reducerPath,MovieApi.reducerPath,torrentApi.reducerPath,trailerApi.reducerPath,fetchCommentApi.reducerPath,auth.reducerPath]
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
  reducer: persistedReducer,
    middleware: (getDefaultMiddleware) => {
      return getDefaultMiddleware({
        serializableCheck: {
          ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
        },
      }).concat(infoTag.middleware,info.middleware,torrentApi.middleware,auth.middleware,MovieApi.middleware,trailerApi.middleware,fetchCommentApi.middleware);
    },
});

setupListeners(store.dispatch)

export const persiter = persistStore(store);
export type TypeRootState = ReturnType<typeof store.getState>;

export type RootState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;

export default store;
