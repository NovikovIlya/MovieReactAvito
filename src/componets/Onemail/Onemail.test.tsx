import React from 'react';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import Onemail from './Onemail';
import { Provider } from 'react-redux';
import store from '../../store';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';



describe('test component mail', () => {
  it('renders spinner', async () => {
    const mail = render(
      <Provider store={store}>
        <BrowserRouter>
          <Onemail />
        </BrowserRouter>
      </Provider>,
    );

    expect(mail).toMatchSnapshot();

    const btn = screen.getByTestId('z');

    expect(btn).toBeInTheDocument();
  });


});