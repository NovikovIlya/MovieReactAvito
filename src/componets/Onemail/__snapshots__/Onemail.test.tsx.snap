// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component mail renders spinner 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ant-spin-nested-loading css-dev-only-do-not-override-i0102m"
      >
        <div>
          <div
            aria-busy="true"
            aria-live="polite"
            class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text zagr css-dev-only-do-not-override-i0102m"
          >
            <span
              class="ant-spin-dot ant-spin-dot-spin"
            >
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
            </span>
            <div
              class="ant-spin-text"
            >
              Loading
            </div>
          </div>
        </div>
        <div
          class="ant-spin-container ant-spin-blur"
        >
          <div
            class="content"
            data-testid="z"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ant-spin-nested-loading css-dev-only-do-not-override-i0102m"
    >
      <div>
        <div
          aria-busy="true"
          aria-live="polite"
          class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text zagr css-dev-only-do-not-override-i0102m"
        >
          <span
            class="ant-spin-dot ant-spin-dot-spin"
          >
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
          </span>
          <div
            class="ant-spin-text"
          >
            Loading
          </div>
        </div>
      </div>
      <div
        class="ant-spin-container ant-spin-blur"
      >
        <div
          class="content"
          data-testid="z"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
